#!/usr/bin/env python

PACKAGE = "yocs_velocity_smoother"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# gen.add("velocity_smoother_on", bool_t, 0, "Velocity Smoother Enabled", True)
gen.add("speed_lim_vx", double_t, 0, "Maximum linear velocity", 1.0, 0.0, 10.0)
gen.add("speed_lim_vy", double_t, 0, "Maximum linear velocity", 1.0, 0.0, 10.0)
gen.add("speed_lim_w", double_t, 0, "Maximum angular velocity", 5.0, 0.0, 10.0)

gen.add("accel_lim_vx", double_t, 0, "Maximum linear acceleration", 0.5, 0.0, 10.0)
gen.add("accel_lim_vy", double_t, 0, "Maximum linear acceleration", 0.5, 0.0, 10.0)
gen.add("accel_lim_w", double_t, 0, "Maximum angular acceleration", 2.5, 0.0, 10.0)


gen.add("decel_lim_vx", double_t, 0, "Maximum linear acceleration", 0.5, 0.0, 10.0)
gen.add("decel_lim_vy", double_t, 0, "Maximum linear acceleration", 0.5, 0.0, 10.0)
gen.add("decel_lim_w", double_t, 0, "Maximum angular acceleration", 2.5, 0.0, 15.0)

#gen.add("decel_factor", double_t, 0, "Deceleration to acceleration ratio", 1.0, 0.0, 10.0)

# Second arg is node name it will run in (doc purposes only), third is generated filename prefix
exit(gen.generate(PACKAGE, "velocity_smoother_configure", "params"))
